.github/workflows/docker-publish.yml

```diff
@@ -0,0 +1,87 @@
+name: Release Docker image to GHCR
+
+on:
+  # TODO: wire into release process
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version to release'
+        required: true
+        default: 'latest'
+
+env:
+  REGISTRY: ghcr.io
+
+jobs:
+  build:
+    permissions:
+      contents: read
+      packages: write
+    strategy:
+      matrix:
+        runner:
+          - ubuntu-22.04-medium-arm64
+          - ubuntu-22.04
+    runs-on: ${{ matrix.runner }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          fetch-tags: true
+      - name: Log in to the Container registry
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Build and push image by digest
+        uses: docker/build-push-action@v6
+        id: build
+        with:
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          labels: |
+            org.opencontainers.image.source=${{ github.repositoryUrl }}
+          outputs: type=image,name=${{ env.REGISTRY }}/${{ github.repository }},push-by-digest=true,name-canonical=true,push=true
+      - name: Export digest
+        run: |
+          mkdir -p /tmp/digests
+          digest="${{ steps.build.outputs.digest }}"
+          touch "/tmp/digests/${digest#sha256:}"
```

Comment:
If I understand correctly this will push a container that has to be referenced by the same sha as the commit from which it's built, right?

Reply:
I think the arch-specific images get pushed to the registry without an explicit label. The digest you see here is a way to refer to those arch-specific images. It would not be the same as the git commit SHA.

The merge action then picks up on the digests, and references them all when making the arch-independent image

Reply:
Its very possible that there is a more efficient way of passing information between the jobs than this